Sources/AppcuesKit/Presentation/Traits/Appcues/DialogContainerViewController.swift

```diff
@@ -53,11 +71,31 @@ internal class DialogContainerViewController: UIViewController {
         case "top":
             containerView.dialogView.topAnchor.constraint(equalTo: containerView.safeAreaLayoutGuide.topAnchor).isActive = true
         case "bottom":
-            containerView.dialogView.bottomAnchor.constraint(equalTo: containerView.safeAreaLayoutGuide.bottomAnchor).isActive = true
+            let constraint = containerView.dialogView.bottomAnchor.constraint(equalTo: containerView.safeAreaLayoutGuide.bottomAnchor)
+            constraint.isActive = true
+            bottomConstraint = constraint
         default:
             containerView.dialogView.centerYAnchor.constraint(equalTo: containerView.centerYAnchor).isActive = true
         }
 
         return self
     }
+
+    @objc
+    private func adjustForKeyboard(notification: Notification) {
+        guard let keyboardValue = notification.userInfo?[UIResponder.keyboardFrameEndUserInfoKey] as? NSValue else { return }
+
+        if notification.name == UIResponder.keyboardWillHideNotification {
+            self.bottomConstraint?.constant = 0
+        } else {
+            switch containerView.dialogView.keyboardAvoidanceStrategy(keyboardFrameInScreen: keyboardValue.cgRectValue) {
+            case .scroll:
+                // The ExperienceStepViewController handles this case and scrolls the content to avoid the keyboard.
+                break
+            case .move(let keyboardFrame):
+                // Not enough room to scroll the firstResponder into view, so adjust the position of the dialogView.
+                self.bottomConstraint?.constant = -keyboardFrame.height
+            }
```

Comment:
curious, how does this still give padding above the keyboard and not just sit directly on top of the keyboard frame like in the previous video in slack?

Reply:
I'm using `keyboardFrame.height` here instead of `intersection.height`. The intersection is smaller because the safe area at the bottom is excluded. I was using `intersection` because that's what I copied from the scrolling implementation (which correctly uses `intersection`), but that's wrong in this case.

Reply:
And regarding landscape: it's not pretty (landscape rarely is), but it works as expected.